plugins/Microsoft.DocAsCode.Build.OperationLevelRestApi/SplitRestApiToOperationLevel.cs

```diff
@@ -0,0 +1,198 @@
+                    Tags = new List<RestApiTagViewModel>(),
+                    Metadata = MergeMetadata(root.Metadata, child.Metadata)
+                };
+
+                // Reset child's uid to "originalUid/operation", that is to say, overwrite of original Uid will show in operation page.
+                child.Uid = string.Join("/", child.Uid, "operation");
+
+                // Reset html id, which is set by template
+                child.HtmlId = null;
+
+                // Reset child's additional content, which will show in operation page.
+                child.Conceptual = null;
+                child.Description = null;
+                child.Summary = null;
+                child.Remarks = null;
+
+                yield return model;
+            }
+        }
+
+        private FileModel GenerateNewFileModel(FileModel model, RestApiRootItemViewModel operationModel)
+        {
+            var originalFile = model.FileAndType.File;
+            var fileExtension = Path.GetExtension(originalFile);
+
+            // When split into operation for petstore.swagger.json, the operation file path should be petstore/{operationName}.json, to prevent operation name confliction
+            var originalFileName = Path.GetFileName(originalFile);
+            var subDirectory = originalFileName.Remove(originalFileName.IndexOf('.'));
+            var directory = Path.GetDirectoryName(originalFile);
+            var filePath = Path.Combine(directory, subDirectory, operationModel.Name + fileExtension).ToNormalizedPath();
+
+            var newFileAndType = new FileAndType(model.FileAndType.BaseDir, filePath, model.FileAndType.Type, model.FileAndType.SourceDir, model.FileAndType.DestinationDir);
+            var newKey = "~/" + RelativePath.GetPathWithoutWorkingFolderChar(filePath);
+            var newModel = new FileModel(newFileAndType, operationModel, model.OriginalFileAndType, model.Serializer, newKey)
+            {
+                LocalPathFromRoot = model.LocalPathFromRoot,
+                Uids = new[] { new UidDefinition(operationModel.Uid, model.LocalPathFromRoot) }.ToImmutableArray()
+            };
+
+            return newModel;
+        }
+
+        private TreeItem ConvertToTreeItem(RestApiRootItemViewModel root, string fileKey)
+        {
+            return new TreeItem()
+            {
+                Metadata = new Dictionary<string, object>()
+                {
+                    [Constants.PropertyName.Name] = root.Name,
+                    [Constants.PropertyName.TopicUid] = root.Uid
+                }
+            };
+        }
+
+        private Dictionary<string, object> MergeMetadata(Dictionary<string, object> rootMetadata, Dictionary<string, object> childMetadata)
+        {
+            var result = new Dictionary<string, object>(rootMetadata);
+            foreach (var pair in childMetadata)
+            {
+                // Root metadata wins for the same key
```

Comment:
Yes I agree

---
In reply to: [127872089](https://github.com/dotnet/docfx/pull/1872#discussion_r127872089) [](ancestors = 127872089)